repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -55,518 +54,373 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
-              }
 
-            processingResult match {
-              case Success(_) => Pass
-              case Failure(e: MissingAuthTokenException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: NotFoundException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: InvalidTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: UnparseableTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: IdentityCommunicationException) => Reject(SC_BAD_GATEWAY, Some(e.getMessage))
-              case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
-              case Failure(e: OverLimitException) =>
-                response.addHeader(HttpHeaders.RETRY_AFTER, e.retryAfter)
-                if (isSelfValidating) {
-                  Reject(e.statusCode, Some(e.getMessage))
-                } else {
-                  Reject(SC_SERVICE_UNAVAILABLE, Some(e.getMessage))
-                }
-              case Failure(e) if e.getCause.isInstanceOf[AkkaServiceClientException] && e.getCause.getCause.isInstanceOf[TimeoutException] =>
-                Reject(SC_GATEWAY_TIMEOUT, Some(s"Call timed out: ${e.getMessage}"))
-              case Failure(_: AdminTokenUnauthorizedException) if isSelfValidating =>
-                Reject(SC_UNAUTHORIZED, Some("Token unauthorized"))
-              case Failure(e) => Reject(SC_INTERNAL_SERVER_ERROR, Some(e.getMessage))
-            }
-          }
+  override def doWork(servletRequest: HttpServletRequest, servletResponse: HttpServletResponse, chain: FilterChain): Unit = {
+    val response = new HttpServletResponseWrapper(servletResponse, MUTABLE, PASSTHROUGH)
 
-        filterResult match {
-          case Pass =>
-            logger.trace("Processing completed, passing to next filter or service")
-            addIdentityStatusHeader(confirmed = true)
-            chain.doFilter(request, response)
-          case Reject(statusCode, message) =>
-            Option(config.getDelegating) match {
-              case Some(delegating) =>
-                logger.debug(s"Delegating with status $statusCode caused by: ${message.getOrElse("unspecified")}")
-
-                val delegationHeaders = buildDelegationHeaders(statusCode,
-                  "keystone-v2",
-                  message.getOrElse("Failure in the Keystone v2 filter").replace("\n", " "),
-                  delegating.getQuality)
-
-                addIdentityStatusHeader(confirmed = false)
-                delegationHeaders foreach { case (key, values) =>
-                  values foreach { value =>
-                    request.addHeader(key, value)
-                  }
-                }
+    super.doWork(servletRequest, response, chain)
 
-                chain.doFilter(request, response)
+    val keystoneAuthenticateHeader = s"Keystone uri=${configuration.getIdentityService.getUri}"
```

Comment:
Seems like we may as well move this inside of the conditional block.

Reply:
yep, it was a holdover of things getting moved around.